src/ModelContextProtocol.Core/UriTemplate.cs

```diff
@@ -453,4 +454,52 @@ static void AppendHex(ref DefaultInterpolatedStringHandler builder, char c)
             }
         }
     }
+
+    /// <summary>
+    /// Defines an equality comparer for Uri templates as follows:
+    /// 1. Non-templated Uris use regular System.Uri equality comparison (host name is case insensitive).
+    /// 2. Templated Uris use regular string equality.
+    /// 
+    /// We do this because non-templated resources are looked up directly from the resource dictionary
+    /// and we need to make sure equality is implemented correctly. Templated Uris are resolved using
+    /// linear traversal so there's no need for equality comparison to be fully accurate.
```

Comment:
I'm not understanding how this statement relates to using this comparer with the ConcurrentDictionary in the collection...?

Reply:
It relates to this fallback, which is where any URI derived via template expansion is meant to be found:

https://github.com/modelcontextprotocol/csharp-sdk/blob/283a052e3f8aa0df3ee6f39f6597f5f0519c9d2e/src/ModelContextProtocol.Core/Server/McpServer.cs#L293-L302

Reply:
> so there's no need for equality comparison to be fully accurate.

But this comparer isn't use at all in that situation, right? I think I'm thrown off by the comment; it's not that the comparison doesn't need to be fully accurate, it's that it doesn't need to be at all (for that use).

Could you clarify in the comment where/when this comparer is used?